### ISSUE TITLE
Preference Model Based Filter Optimization

### ISSUE THREAD

**Issue by jaakkopasanen** [jaakkopasanen/AutoEq#35]
Currently filters are optimized with loss function that only takes the average deviation into account. While this works alright for cases where there are plenty of parametric eq filters available less flexible systems are not necessarily optimized well.

Prime example of this is Sennheiser HD 800 with standard 10-band equalizer. The main problem of HD 800 is it's 6 kHz peak which happens to fall right in the middle of 4 kHz and 8 kHz filters. Simple filter optimization cannot reduce gain on 4 kHz and 8 kHz filters because that would affect other frequencies as well. However not all deviations from target curve are equal. A peak in treble is perceived to have greater negative effect than having a similar notch.

![HD 800](https://raw.githubusercontent.com/jaakkopasanen/AutoEq/master/results/oratory1990/harman_over-ear_2018/Sennheiser%20HD%20800/Sennheiser%20HD%20800.png)

Ideally filter optimization would be based on preference model like the one developed by Sean Olive and others for their 2018 Harman over-ear target curve. This is obviously only defined for Harman target compatible measurement systems but probably would be significant improvement over current simple model on other measurements as well if the chosen target curve is assumed to represent neutral.

**Comment by jaakkopasanen** [comment]
Looks like Harman target preference model doesn't have different weights for having peak vs notch in the same treble frequency. Many would prefer to have 2 dB too little gain in 6 kHz than 2 dB too much.

This whole problem is only relative to 10-band equalizers because parametric eq and FIR filters are flexible enough to produce good enough equalization curves. Problem with 10-band equalizer can be solved by each used because it's so easy to adjust the filter gains manually. I'll add a note in the usage instructions to actually listen to the headphone when using 10-band eq and adjust the filters close to the problem manually.

**Comment by jaakkopasanen** [comment]
Would be cool to run some experiments training a neural network regressor for predicting the preference score. Training data is pairs of frequency responses and a label that says which one is preferred. In the beginning the lables can be simulated with some other preference model like the Harman one. This would then serve as a tool for training a model with data gathered from actual listening tests.

Simply predicting the label won't serve for single frequency response preference score because the labels are done by comparing two. So there needs to be a some kind of transformation to get the actual regeression values. Maybe generate huge amount of frequency responses, compare them against each other and sort by the preference label. This sorting weighted with label prediction confidences would serve as the basis for the preference score.

Reason for going with this method is to make it easier for the listeners. Saying which of two frequency responses is better is hard enough let alone telling how much better it is.
